src/main/java/svc/managers/SMSManager.java

```diff
@@ -0,0 +1,248 @@
+import com.twilio.twiml.MessagingResponse;
+
+import svc.dto.CitationSearchCriteria;
+import svc.managers.ViolationManager.VIOLATION_STATUS;
+import svc.models.Citation;
+import svc.models.Court;
+import svc.models.TwimlMessageRequest;
+import svc.models.Violation;
+
+@Component
+public class SMSManager {
+	@Inject
+	CitationManager citationManager;
+	@Inject
+	private CourtManager courtManager;
+	@Inject
+	private ViolationManager violationManager;
+	
+	private enum SMS_STAGE{
+		WELCOME(0),
+		READ_DOB(1),
+		READ_LICENSE(2),
+		VIEW_CITATION(3);
+		
+		private int numVal;
+		
+		SMS_STAGE(int numVal){
+			this.numVal = numVal;
+		}
+		
+		public int getNumVal(){
+			return numVal;
+		}
+	}
+	
+	
+	//a null String means validDOB, otherwise a message is returned to pass on to user
+	private String validDOB(String dob){
+		String errMsg = "";
+		dob = dob.trim();
+		if (!dob.matches("^\\d{2}([./-])\\d{2}\\1\\d{4}$")){
+			errMsg = "Please re-enter your birthdate.  The format is: MM/DD/YYYY";
+		}else{
+			if (isValidDate(dob)){
+				//Check that the DOB is > 18 years old
+				DateTimeFormatter formatter = DateTimeFormat.forPattern("MM/dd/yyyy");
+				DateTime dobDT = formatter.parseDateTime(dob);
+				DateTime now = new DateTime();
+				Years age = Years.yearsBetween(dobDT, now);
+				if (age.getYears() < 18){
+					errMsg = "You must be at leat 18 years old to use www.yourSTLcourts.com";
+				}
+			}else{
+				//DOB is not a valid date
+				errMsg = "The date you entered: '"+dob+"' was not a valid date.  Please re-enter your birthdate using MM/DD/YYYY";
+			}
+		}
+		
+		return errMsg;
+	}
+	
+	
+	private boolean isValidDate(String dateToCheck){
+		SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+	    dateFormat.setLenient(false);
+	    try {
+	      dateFormat.parse(dateToCheck.trim());
+	    } catch (ParseException pe) {
+	      return false;
+	    }
+	    return true;
+	}
+	
+	private String convertDatabaseDateToUS(Date databaseDate){
+		SimpleDateFormat usFormat = new SimpleDateFormat("MM/dd/yyyy");
+		
+		return usFormat.format(databaseDate);
+	}
+	
+	private String getViolationSMS(Violation violation, boolean showDismissedViolations){
+		String message = "";
+		//by default Closed violations will not be returned.
+		VIOLATION_STATUS status = VIOLATION_STATUS.valueOf(violation.status.replaceAll(" ", "_"));
+		if ((status != VIOLATION_STATUS.CLOSED) && (status != VIOLATION_STATUS.DISMISS_WITHOUT_COSTS || showDismissedViolations)){
+			message += "\nViolation #: "+violation.violation_number;
+			message += "\nViolation: "+violation.violation_description;
+			message += "\nStatus (as of "+convertDatabaseDateToUS(violation.status_date)+"): "+status.toString();
+			if (status != VIOLATION_STATUS.DISMISS_WITHOUT_COSTS){
+				message += "\nFine Amount: $"+violation.fine_amount;
+				message += "\nCourt Costs: $"+violation.court_cost;
+			}
+		}
+		return message;
+	}
+	
+	public String getCitationSMS(Citation citation){
```

Comment:
It might be cleaner/easier to maintain, to build representation objects for each 'text template' we want to send back to the user (in this case CitationTextMessage) that takes a citation in, and has a 'ToTextMessage()' function that builds the string or a CitationTextMessageBuilder that has a fromCitation(citation) function. Just a thought.